### ISSUE TITLE
Regex composition library

### ISSUE THREAD

**Issue by Makman2** [coala/coala#3882]
For example linters often need to match paths, so a library that allows to compose more complex regexes could be beneficial:

```
regex_composition.path + r': error found'
```
(for example `path` could also change depending on the OS, like Windows vs Linux for more consistent behaviour)

This should probably go into a separate repository^^

**Comment by Makman2** [comment]
Different kind of links may be also worth to consider :+1:

**Comment by bhavishyagopesh** [comment]
@Makman2 I woud like to take this up.

**Comment by meetmangukiya** [comment]
@bhavishyagopes, ok, lets discuss first, then(since it has `status/needs design` label)

**Comment by Mariatta** [comment]
@Makman2 @meetmangukiya , @bhavishyagopesh  is interested to work on this issue as part of gsoc. I understand it is still need design and consideration. Please guide @bhavishyagopesh with this if you can. Thanks.

**Comment by Makman2** [comment]
Not sure this really fits into a gsoc (except when doing some kind of allround work). Maybe there's already another python package that contains those features, then we could add stuff there. This has to be researched once and then we would slowly start porting regexes / use existing regexes.

**Comment by bhavishyagopesh** [comment]
@Makman2 So what do you suggest here do I look out for something(i.e. python package for generating complex regexes).

**Comment by Makman2** [comment]
yes look for something existent and if it suits our needs. You can grep in our repo for regexes we use so you see what we use/need ;) If you can't find a package that goes into the right direction (but I think there are ones), we make a new repo. But if possible consider preparing upstream contributions :+1:

**Comment by bhavishyagopesh** [comment]
For now I found [this](http://zesty.ca/python/rxb.py), and I'm looking for what we need.

**Comment by bhavishyagopesh** [comment]
@Makman2 Still I'm not very sure of which files will undergo change, say using this script we can redefine regexes in  coala/coalib/misc/Constants.py  but changes have to made everywhere, so now could exactly point  in what direction should I go.

**Comment by Makman2** [comment]
grep for modules that use `re` and look for regexes inside^^

**Comment by bhavishyagopesh** [comment]
Yeah,I have done that, do you want to say that script doesnt has enough functions...

**Comment by Makman2** [comment]
I'm saying nothing, haven't looked at the script^^

**Comment by bhavishyagopesh** [comment]
Please if you could elaborate(I desperately need a patch submission to back my project proposal) and (also what does ^^ mean,googled but didnt got decisive answers. )....I greped and the functions in the script can be used to generate most of them(I suppose).

**Comment by Makman2** [comment]
Then I would advise you to take an issue with `difficulty/newcomer`/`difficulty/low`, this one's still marked as `needs-design` as there needs some investigation on this issue.

**Comment by jayvdb** [comment]
Using [`rebulk`](http://pypi.python.org/pypi/rebulk) or similar existing libraries would be better than creating our own.
